operator/src/main/java/oracle/kubernetes/operator/helpers/JobHelper.java

```diff
@@ -66,13 +66,58 @@ private static boolean runIntrospector(Packet packet, DomainPresenceInfo info) {
     LOGGER.fine("runIntrospector topology: " + topology);
     LOGGER.fine("runningServersCount: " + runningServersCount(info));
     LOGGER.fine("creatingServers: " + creatingServers(info));
-    return topology == null || isBringingUpNewDomain(info);
+    return topology == null || isBringingUpNewDomain(info) || isModelInImageUpdate(packet, info);
   }
 
   private static boolean isBringingUpNewDomain(DomainPresenceInfo info) {
     return runningServersCount(info) == 0 && creatingServers(info);
   }
 
+  private static boolean isModelInImageUpdate(Packet packet, DomainPresenceInfo info) {
+    if (info.getDomain().getDomainHomeSourceType().equals("FromModel")) {
+
+      String currentPodRestartVersion = info.getDomain().getAdminServerSpec().getDomainRestartVersion();
+      String currentPodIntrospectVersion = info.getDomain().getAdminServerSpec().getDomainIntrospectVersion();
+      String configMapRestartVersion = (String)packet.get(ProcessingConstants.DOMAIN_RESTART_VERSION);
+      String configMapIntrospectVersion = (String)packet.get(ProcessingConstants.DOMAIN_INTROSPECT_VERSION);
+      String configMapDomainName = (String)packet.get(ProcessingConstants.DOMAIN_INPUTS_HASH);
+      String currentImageName = info.getDomain().getSpec().getImage();
+
+      LOGGER.finest("JobHelper.isModelInImageUpdate currentPodRestartVersion " + currentPodRestartVersion);
+      LOGGER.finest("JobHelper.isModelInImageUpdate currentPodIntrospectVersion " + currentPodIntrospectVersion);
+      LOGGER.finest("JobHelper.isModelInImageUpdate configMapRestartVersion " + configMapRestartVersion);
+      LOGGER.finest("JobHelper.isModelInImageUpdate configMapIntrospectVersion " + configMapIntrospectVersion);
+
+      // If either one is set, check for differences and decide to run intropsect job
+
+      if (currentPodIntrospectVersion != null
+            && !currentPodIntrospectVersion.equals(configMapIntrospectVersion)) {
+        return true;
+      }
+
+      if (currentPodRestartVersion != null
+            && !currentPodRestartVersion.equals(configMapRestartVersion)) {
+        return true;
+      }
+
+      if (configMapRestartVersion != null
+          && !configMapRestartVersion.equals(currentPodRestartVersion)) {
+        return true;
+      }
+
+      if (configMapIntrospectVersion != null
+          && !configMapIntrospectVersion.equals(currentPodIntrospectVersion)) {
+        return true;
+      }
+
+      if (!currentImageName.equals(configMapDomainName)) {
```

Comment:
I may have missed something, but this comparison looks wrong to me.

1) The configMapDomainName is the string value of the hashcode of the image name in the domain spec. See line #83 in this class, line#400 and #410 in ConfigMapHelper, and the getModelInImageSpecHash() method in DomainPresenceInfo. But we are comparing it with the current image name itself. 
2) Potentially we may add more domain spec attributes into getModelInImageSpecHash() calculation. Once we do that, comparing a hashCode of a set of attributes with one attribute (or its hashcode) alone may not be correct.

Maybe we want to use the hashcode of the current spec, instead the image name, to do the comparison(?).

Reply:
Yes, it should be the hash of the image name 

But model in image, we don't want to use the hashcode for the entire spec ,  some attributes should not trigger the introspect job

Reply:
Sorry, I did not mean the hashcode of the whole spec. I meant to call the result of the same getModelInImageHash method for the new and current spec.

Reply:
I refactored the code and created the static method in ConfigMapHelper

 public static int getModelInImageSpecHash(String imageName) and also corrected the naming and logic.

Reply:
IS this fixed? If so, please close the comment.